discovery_engine_core/core/src/engine.rs

```diff
@@ -558,12 +558,47 @@ impl Engine {
     }
 
     /// Process the feedback about the user spending some time on a document.
-    pub async fn time_spent(&mut self, time_spent: &TimeSpent) {
-        if let UserReaction::Positive | UserReaction::Neutral = time_spent.reaction {
+    pub async fn time_spent(&mut self, time_spent: TimeSpent) -> Result<(), Error> {
+        cfg_if! {
+            if #[cfg(feature = "storage")] {
+                use storage::models::TimeSpendDocumentView;
+
+                let TimeSpent {
+                    id,
+                    view_time,
+                    view_mode,
+                    // dummy values when storage is enabled
+                    smbert_embedding: _,
+                    reaction: _,
+                } = time_spent;
+
+                let TimeSpendDocumentView {
+                    smbert_embedding,
+                    aggregated_view_time,
+                    last_reaction,
+                } = self
+                    .storage
+                    .feedback()
+                    .update_time_spend(id, view_mode, view_time)
+                    .await?;
+
+                let last_reaction = last_reaction.unwrap_or(UserReaction::Neutral);
```

Comment:
🔧 the default could be used directly
```suggestion
                let last_reaction = last_reaction.unwrap_or_default();
```